scripts/build_vcpkg_deps.ps1

```diff
@@ -0,0 +1,173 @@
+    [Parameter(Mandatory=$true)][bool]$BuildAnimView,
+    [Parameter(Mandatory=$true)][bool]$IsX64Build,
+    [Parameter(Mandatory=$true)][string]$VcpkgCommitSha
+)
+
+################
+# Variables
+################
+
+$anim_view_libs = "wxwidgets"
+$corsixth_libs = "ffmpeg", "freetype", "lua", "luafilesystem", "lpeg", "sdl2", "sdl2-mixer"
+
+$vcpkg_git_url = "https://github.com/CorsixTH/vcpkg"
+
+$dest_folder_name = "vcpkg"
+$dest_folder_path = ".\vcpkg"
+
+$x86_triplet_name = "x86-windows"
+$x64_triplet_name = "x64-windows"
+
+###################
+# Functions
+###################
+
+function run_command($command){
+    Invoke-Expression $command
+    if ($LASTEXITCODE -ne 0){
+        throw "Failed to run command :`n $command `nExiting."
+    }
+}
+
+##################
+# Main script
+##################
+
+# Wrap in function so we can try-catch
+function run_script {
+
+    # Test the required files are in the path
+    if ((Get-Command "git.exe") -eq $null) {
+        throw "Error git was not found. Is it installed, added to your path 
+               and have you restarted since?"
+    }
+
+    # Check we have the latest copy of vcpkg
+    if (-Not (Test-Path $dest_folder_path)){
+        # If vcpkg does not exist clone it
+        run_command -command "git clone $vcpkg_git_url $dest_folder_name"
+        Set-Location -Path $dest_folder_path
+    } else {
+        # Move into vcpkg folder and update to latest version
+        Set-Location -Path $dest_folder_path
+        run_command "git reset --hard; git fetch origin; git checkout $VcpkgCommitSha"
+    }
+
+    $commit_id_filename = "commit_id.txt"
+    if (-Not (Test-Path $commit_id_filename)){
+        # Assume we have not build this dir before, bootstrap to start.
+        run_command "git clean -fx"
+        Write-Output "Previous sha not found, cleaning and bootstrapping vcpkg."
+        run_command ".\bootstrap-vcpkg.bat"
+        Set-Content -Path $commit_id_filename -Value $VcpkgCommitSha
+    }
+
+    # File exists so check the SHA we last saw
+    $previous_sha = Get-Content $commit_id_filename
+    if ($previous_sha -ne $VcpkgCommitSha){
+        # SHA has changed
+
+        # Check if we have gone back in time first by checking if
+        # previous sha was known at this point
+        $commit_history_command = "git merge-base --is-ancestor $previous_sha $VcpkgCommitSha "
+        Invoke-Expression -command $commit_history_command
+
+        if ($LASTEXITCODE -ne 0){
+            # Git did not recognise this commit id. We have gone backwards
+            # completely clean before continuing
+            Write-Output "Dependencies has been rolled back. Cleaning for safety."
+            run_command "git clean -fx"
+            # Bootstrap again
+            run_command ".\bootstrap-vcpkg.bat"
+        } else {
+            # We have moved forwards, we can use normal mechanism
+            Write-Output "Dependencies have been updated."
+            run_command ".\vcpkg update"
```

Comment:
Unfortunately at this time `vcpkg update` doesn't actually perform an update, it just finds a list of ports that have updates and tells you to remove and install them again. I don't know how fancy you want to get here.

Reply:
Okay lets save ourselves lots of headaches and just always clean build dependencies if we change the commit that we point to.

Reply:
I think you would also be ok with `./vcpkg remove --outdated` then sending the whole list to install. On the train though so I can't test.

Reply:
Checking the vcpkg source it sadly looks like this isn't allowed. So sure, clean and rebuild.

Reply:
I don't have total context, but hopefully our design is that you can simply do
```
./vcpkg remove --outdated --recurse
./vcpkg install <list of all dependencies>
```
Any deps that are still installed will be a no-op, any deps that got removed will be installed.

Reply:
@ras0219-msft - Would these commands also work if we rolled back to a specific commit or would we have to take the clean build approach?

Reply:
They will still work, though you should re-bootstrap the tool as well.
```
.\bootstrap-vcpkg.bat
```
We treat versions as "cookies" and don't try to say less-than/greater-than. If it's different, it's wrong :).

Reply:
Great thanks for the heads up, this makes it significantly easier. 

Could we add this to the documentation in the vcpkg repository? 
The only details I could find about updating was [here](https://github.com/Microsoft/vcpkg/blob/master/docs/about/faq.md#how-do-i-update-libraries). Beyond this I couldn't find details that you've provided such as re-running the bootstrap script or how versions are handled when I was looking.